crypto/src/signature/secp256k1.rs

```diff
@@ -211,15 +201,14 @@ mod test {
 
     #[test]
     fn secp256k1_sign() {
-        let secret = PrivateKey::from_hex(ALGORITHM, PRIVATE_KEY).unwrap();
-        let (pk, sk) =
-            EcdsaSecp256k1Sha256::keypair(Some(KeyGenOption::FromPrivateKey(secret))).unwrap();
+        let secret = private_key();
+        let (pk, sk) = EcdsaSecp256k1Sha256::keypair(Some(KeyGenOption::FromPrivateKey(
+            crate::PrivateKey::Secp256k1(secret),
+        )));
 
-        let sig = EcdsaSecp256k1Sha256::sign(MESSAGE_1, &sk).unwrap();
+        let sig = EcdsaSecp256k1Sha256::sign(MESSAGE_1, &sk);
         let result = EcdsaSecp256k1Sha256::verify(MESSAGE_1, &sig, &pk);
-        // we are returning a `Result<bool>`
-        // unwrap will catch the `Err(_)`, and assert will catch the `false`
-        assert!(result.unwrap());
+        assert!(result.is_ok());
```

Comment:
I think just an unwrap will do and will give a marginally better experience (it will print the error message).

Some other tests in `iroha_crypto` also use the `assert!(result.is_ok())` idiom, but I don't think it should be used ever, even if the error message is non-helpful (like in the crypto stuff), it won't tempt you to look at it

Reply:
Fixed in all `crypto/` code.

I'm kind off dissapointed that there is no idiom to express the logic of *assertions being tested* and *assertions which just expected to work in order to test the first ones*.

That can probably be expressed with panicking for first and with `?` for second, but it's not a common practice as I can see